browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5573,9 +5573,13 @@ L.CanvasTileLayer = L.Layer.extend({
 		this._debugTyperTimeoutId = setTimeout(L.bind(this._debugTyperTimeout, this), 50);
 	},
 
-	_debugTypeText: function(text) {
+	_debugTypeText: function(text, delayMs) {
 		for (var i=0; i<text.length; i++) {
-			this._debugTypeChar(text.charCodeAt(i));
+			if (delayMs) {
+				setTimeout(L.bind(this._debugTypeChar, this, text.charCodeAt(i)), i*delayMs);
```

Comment:
Looks better, but how do we chain these things helpfully ? promises or something clever ? I guess the latest JS async support is not available in all browser and will not parse and/or fall in a heap for older users but - ... =)